### ISSUE TITLE
A small number of hosts/services became stale after reload on config master

### ISSUE THREAD

**Issue by jhou4** [Icinga/icinga2#7241]
## Describe the bug

Occasionally, a small number of hosts/services became stale on config master right after reload. Those hosts/services are in the same zone. Other services/hosts in the same zone, or other zones, are fine. The two satellites in that zone, and the other master do have those hosts/services status constantly updated in databases. |
Restarting icinga2 service on config master did not help. But restarting icinga2 service on two satellites in that zone fixed the issue. 

## To Reproduce

I am not aware of a reliable way to reproduce this issue. 

## Expected behavior

Hosts/services status are updated on all nodes in the cluster.

## Screenshots
I did not collect screenshots before fixing it. But I can capture a picture next time it happens if needed. 

## Your Environment

Version used (icinga2 --version):
r2.10.4-1

Operating System and version:
Red Hat Enterprise Linux Server release 7.3 (Maipo)

Enabled features (icinga2 feature list):
Satellites: Enabled features: api checker mainlog
Masters:   Enabled features: api checker command compatlog mainlog notification statusdata

Icinga Web 2 version and modules (System - About):
2.6.2
monitoring | 2.6.2
setup | 2.6.2

Config validation (icinga2 daemon -C):
...
[2019-06-12 14:08:22 -0400] information/cli: Finished validating the configuration file(s).

If you run multiple Icinga 2 instances, the zones.conf file (or icinga2 object list --type Endpoint and icinga2 object list --type Zone) from all affected nodes.

object Zone "global-templates" {
global = true
}

object Zone "master" {
endpoints = ["master1", "master2"]
}

object Zone "zone1" {
endpoints = ["sat1", "sat2"]
parent = "master"
}

object Zone "zone2" {
endpoints = ["sat3", "sat4"]
parent = "master"
}

## Additional context

I can provide additional info as needed.

**Comment by dnsmichi** [comment]
Sounds like a deadlock or lockup on the satellite nodes, did you check their logs for any problems? Also, a full gdb backtrace from the hanging satellites would help provide more insights, details on this can be found in the development chapter in the docs.

**Comment by jhou4** [comment]
I did check icinga2 logs on master and two satellites, did not see anything abnormal. But I think we found the issue - We did not check the second master at the time but later we found the configuration(/var/lib/icinga2/api/zones) on the second master was out of sync for some reason. After we cleaned up /varlib/icinga2/api and restarted icinga2 on the second master, problem went away. 

So the cause appears to be configuration sync within the cluster. Since we know the workaround, we can close this issue. But if it is possible to improve cluster configuration sync, it would be better.

Thanks.
